src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs

```diff
@@ -111,6 +111,9 @@ public override int GetHashCode()
 
                 public override bool Equals(Diagnostic? obj)
                 {
+                    if (obj is CompilerDiagnostic other)
+                        return _original.Equals(other._original);
+
                     return _original.Equals(obj);
```

Comment:
>_original.Equals(obj) [](http://example.com/codeflow?start=27&length=21)

Presumably the existing code was handling wrapped == unwrapped. Are we handling unwrapped == wrapped? If we need to handle that case, that may be another reason to remove this type instead.

Reply:
Agree. It looks like equality is not symmetrical.

Reply:
Thanks, I have adjusted the check so we only support comparing two instances of _wrapped_ diagnostics. Any host that wants to compare wrapped against unwrapped should use a custom comparer.